plugin/plugin.go

```diff
@@ -177,7 +216,60 @@ func StartPlugin(pd *pluginDescriptor, action pluginScope) (*plugin, error) {
 	if len(command) == 0 {
 		return nil, fmt.Errorf("Platform specific command not specified: %s.", runtime.GOOS)
 	}
+	if pd.hasCapability(gRPCSupportCapability) {
+		return startGRPCPlugin(pd, command)
+	}
+	return startLegacyPlugin(pd, command)
+}
+
+func startGRPCPlugin(pd *pluginDescriptor, command []string) (*plugin, error) {
+	portChan := make(chan string)
+	writer := &logger.LogWriter{
+		Stderr: logger.NewCustomWriter(portChan, os.Stderr, pd.ID),
+		Stdout: logger.NewCustomWriter(portChan, os.Stdout, pd.ID),
+	}
+	cmd, err := common.ExecuteCommand(command, pd.pluginPath, writer.Stdout, writer.Stderr)
+	go func() {
+		err = cmd.Wait()
+		if err != nil {
+			logger.Errorf(true, "Error occurred while waiting for plugin process to finish.\nError : %s", err.Error())
+		}
+	}()
+	if err != nil {
+		return nil, err
+	}
+
+	var port string
+	select {
+	case port = <-portChan:
+		close(portChan)
+	case <-time.After(config.PluginConnectionTimeout()):
+		return nil, fmt.Errorf("timed out connecting to %s", pd.ID)
+	}
+	logger.Debugf(true, "Attempting to connect to grpc server at port: %s", port)
+	gRPCConn, err := grpc.Dial(fmt.Sprintf("%s:%s", "127.0.0.1", port),
+		grpc.WithInsecure(),
+		grpc.WithDefaultCallOptions(grpc.MaxCallSendMsgSize(1024*1024*1024*15), grpc.MaxCallRecvMsgSize(1024*1024*1024*15)),
+		grpc.WithBlock())
+	if err != nil {
+		return nil, err
+	}
+	plugin := &plugin{
+		pluginCmd:     cmd,
+		descriptor:    pd,
+		gRPCConn:      gRPCConn,
+		ProcessClient: gauge_messages.NewProcessClient(gRPCConn),
+		ResultClient:  gauge_messages.NewResultClient(gRPCConn),
+	}
+	if plugin.descriptor.hasCapability(executionService) {
+		plugin.ExecutionClient = gauge_messages.NewExecutionClient(gRPCConn)
+	}
+
+	logger.Debugf(true, "Successfully made the connection with runner with port: %s", port)
```

Comment:
Not runner. Made connection with plugin <plugin name> should be printed in logs.